src/osd/ECBackend.cc

```diff
@@ -1008,6 +1008,23 @@ shard_to_read_map.find(from);
   rop.in_progress.erase(from);
   if (!rop.in_progress.empty()) {
     dout(10) << __func__ << " readop not complete: " << rop << dendl;
+    if (subread_all) {
+      int k = ec_impl->get_data_chunk_count();
+      for (map<hobject_t, read_result_t>::iterator i =
+            rop.complete.begin();
+          i != rop.complete.end();
+          ++i) {
+        if ( i->second.returned.front().get<2>().size() < k ) {
+          dout(10) << __func__ << " readop not complete: " << rop << dendl;
+          return;
+        }
```

Comment:
Yeah.. currently we finish the transaction right after getting k response, and clear everything there. And relay on scrubbing for the error detect and correction..